src/HorseProfile/HorseProfile.js

```diff
@@ -16,16 +16,32 @@ function HorseProfile() {
     vet: undefined,
     owner: undefined,
   })
+
   const horseId = Number(useParams().id);
+  const navigate = useNavigate();
 
   const {loading, error} = useQuery(fetchHorse, {
     variables: { id: horseId },
     onCompleted: data => {
       setHorse(() => data.fetchHorse);
-
     }
   })
 
+  const [destroyHorse] = useMutation(destroyHorseQuery, {
+    refetchQueries: [
+      fetchAllHorses, 
+      'fetchHorses' 
+    ],
+  });
+
+  const removeHorse = () => {
+    if (window.confirm('Are you sure you want to delete this horse?')) {
+      destroyHorse({
+        variables: {input: {id: Number(horseId)}}
+      })
+      navigate('/horses')
```

Comment:
This is so cool! I had no idea Router had this hook. I assume this automatically takes you back to the horses page when you delete?